swap/peer.go

```diff
@@ -30,18 +35,144 @@ var ErrDontOwe = errors.New("no negative balance")
 // Peer is a devp2p peer for the Swap protocol
 type Peer struct {
 	*protocols.Peer
+	lock               sync.RWMutex
 	swap               *Swap
 	beneficiary        common.Address
 	contractAddress    common.Address
 	lastReceivedCheque *Cheque
+	lastSentCheque     *Cheque
+	balance            int64
 }
 
 // NewPeer creates a new swap Peer instance
 func NewPeer(p *protocols.Peer, s *Swap, beneficiary common.Address, contractAddress common.Address) *Peer {
-	return &Peer{
+	peer := &Peer{
 		Peer:            p,
 		swap:            s,
 		beneficiary:     beneficiary,
 		contractAddress: contractAddress,
 	}
+
+	peer.lastReceivedCheque, _ = s.loadLastReceivedCheque(p.ID())
+	peer.lastSentCheque, _ = s.loadLastSentCheque(p.ID())
+	peer.balance, _ = s.loadBalance(p.ID())
+
+	return peer
+}
+
+func (p *Peer) getLastReceivedCheque() *Cheque {
+	return p.lastReceivedCheque
+}
+
+func (p *Peer) getLastSentCheque() *Cheque {
+	return p.lastSentCheque
+}
+
+func (p *Peer) setLastReceivedCheque(cheque *Cheque) error {
+	p.lastReceivedCheque = cheque
+	return p.swap.saveLastReceivedCheque(p.ID(), cheque)
+}
+
+func (p *Peer) setLastSentCheque(cheque *Cheque) error {
+	p.lastSentCheque = cheque
+	return p.swap.saveLastSentCheque(p.ID(), cheque)
+}
+
+func (p *Peer) getLastChequeValues() (total uint64, err error) {
```

Comment:
I would probably rename this, in fact there is only one value returned, not multiple. `getLastChequeCumulativeAmount` comes to mind, but maybe `getCumulativeAmount` or something similar is enough?

Reply:
done